src/main/java/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,23 @@
+package service.operation;
+
+import dao.FruitDao;
+
+public class PurchaseOperation implements OperationHandler {
+    private final FruitDao fruitDao;
+
+    public PurchaseOperation(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void handle(String fruit, int quantity) {
+        int balanceAfterPurchase = fruitDao.getBalance(fruit) - quantity;
+        if (balanceAfterPurchase >= 0) {
+            fruitDao.updateBalance(fruit,balanceAfterPurchase);
+        } else {
+            throw new RuntimeException(String.format("You wanted to buy %d %ss "
+                    + "but available only %d %ss",quantity, fruit,fruitDao.getBalance(fruit),fruit
+            ));
+        }
```

Comment:
in cases like this, invert if condition to throw early - "guard clause pattern". We wont'need else after return/throw
https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement